_source/_posts/2020-12-16-react-login.md

```diff
@@ -0,0 +1,599 @@
+
+**TIP**: You can also create an account and register apps using the [Okta CLI](https://github.com/okta/okta-cli#readme) with the `okta register` and `okta apps create` commands.
+
+In your Okta Development Console, select **Applications** from the top menu and click **Add Application**. Next, select **Single-Page App** and click **Next**. You now see a form with application settings. You can give the application any name of your choice. Both the **Base URI** and the **Logout redirect URI** should be changed to `http://localhost:3000/`.
+
+You will be testing the application using the React development server that listens on port 3000 on your local machine. For production, you would of course change this to the URI of the production server. The **Login redirect URI** should be `http://localhost:3000/callback`.
+
+Click **Done** when ready. On the next page, you will see an overview of the settings as well as a **Client ID**. Note the ID for later when you set up your React application.
+
+## Create a React Application for Login
+
+Now you are ready to create the React application. Open a terminal in a folder of your choice and run the following command.
+
+```bash
+npx create-react-app react-login
+```
+
+This will create a new React project in the folder `react-login` and set up all the necessary build infrastructure. Now, navigate into the new directory and install the React router.
+
+```bash
+cd react-login
+npm install -E react-router-dom@5.2.0
+```
+
+The router manages the browser routes and maps them to React components. You will be creating two routes for now. The first route is the homepage of the application. The second route represents a restricted area that should only be available to users that are logged in.
+
+Open an editor of your choice in the project and create a file `src/Home.js`. Paste the code below into the new file.
+
+```jsx
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+function Home() {
+  return (
+    <div className="page">
+      <h1>Login with React</h1>
+      <Link to="/private">
+        <button type="button">
+          Visit Restricted Page
+        </button>
+      </Link>
+    </div>
+  );
+}
+
+export default Home;
+```
+
+This is a simple page containing a button that links to a restricted page. The `Link` component is part of the router and allows you to declare links to different parts of the application. Similarly, create a file `src/Private.js` and paste the following contents into it.
+
+```jsx
+import React from 'react';
+
+function Private() {
+  return (
+    <div className="page">
+      <div className="warning">
+        <h1>Restricted Access</h1>
+        <h2>Authorized Personnel Only</h2>
+      </div>
+    </div>
+  );
+}
+
+export default Private;
+```
```

Comment:
This might not be the right place to mention this, if at all, but I think it's worth pointing out that while this page will be restricted by React, because it's a client-side application, the text here will be visible to any adventurous programmer, so you need to be careful what you put here.